### PR TITLE
Uwsgi graceful restart

### LINKED ISSUE #6: Use supervisor signals to gracefully restart uwsgi
- [x] Use https://github.com/EDITD/ansible-supervisor_task/releases/tag/v0.9 -- done at repo/playbook level, I've not found how one can specify dependency role versions
- [x] Allow for normal restart via var

### PR DESCRIPTION
**Pull request by pkatseas**
@EDITD/hackers This deals with graceful uwsgi restarting. Closes #6

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -1,29 +1,50 @@
...
   sudo: yes
+
+# Optionally, restart the supervisor task group when requested by the user.
+# This does not gracefully restart the uwsgi workers, so not preferable,
+# but it might be needed for certain use cases.
+- supervisorctl:
+    name="{{ app_name }}:"
+    state=restarted
+  sudo: yes
+  when: restart_app
```
Comment (johnspicer) [inline tasks/main.yml]: Do we need this here? If we define `restart_task` then won't it be handled inside `ansible-supervisor_task`?

**Comment by Jberrz** [comment]
LGTM!

File: tasks/main.yml (hunk 1 above)
Comment (pkatseas) [inline tasks/main.yml]: Good point, I added this afterward and didn't link it back to the supervisor_task call.
How's this? 2e5e48d

Comment (johnspicer) [inline tasks/main.yml]: :+1: Yeah I couldn't see a difference between `restart_app` and `restart_task`! Thanks

**Comment by johnspicer** [comment]
LGTM!

**Comment by pkatseas** [comment]
Thanks for the reviews @Jberrz @johnspicer :+1:

**Pull request merged**